geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/Message.java

```diff
@@ -132,9 +132,12 @@
   }
 
   /**
-   * maximum size of an outgoing message. See GEODE-478
+   * The maximum size of an outgoing message. If the message is larger than this maximum, it may
+   * cause the receiver to throw an exception on message part length mismatch due to overflow in
+   * message size.
    */
-  private final int maxMessageSize;
+  private static final int maxMessageSize =
```

Comment:
The downside of making it `static final` is that it becomes untestable. No test can specify or change the value. The alternative would be to move it up a level and just pass `int maxMessageSize` in via the constructor. That keeps this class more testable.

Reply:
The maxMessageSize is currently not being changed in any tests, so I think that keeping it this way makes sense. If we set it in the constructor, we would have the same performance issue as we do currently since the value is being retrieved from a system property, which requires locking. I believe that in order to remove that performance issue, the best solution is the make the value static and initialized at the class level. However since it is a common pattern of ours to make values like that non-static for the purposes of testing, I will add a comment explaining why it is static.

Reply:
I believe there is an annotation for marking it mutable for testing only. 

I agree with both of you. Something is broken here. I am not sure the best way to fix it. Ideally we’d stop doing this stupid system property crap but we just keep adding to that mess. The configuration for this object should come from its parent but that’s the same problem, should it be static there too? Finding a place that is not static but isn’t frequently instantiated so that it can be fetch infrequently would be the best compromise. 

I think I’d like to see this ticket focus on the severe performance problem. And write up another to refactor the relationship such that this doesn’t need to be static but doesn’t get fetch from system properties more than once.